.github/workflows/package-component.yaml

```diff
@@ -0,0 +1,58 @@
+name: Package on Tag
+
+on:
+  push:
+    tags:
+      - '*/*/*/v*.*.*'
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract tag details
+        id: tag_info
+        run: |
+          TAG=${GITHUB_REF#refs/tags/}
+          COMPONENT_TYPE=$(echo $TAG | cut -d/ -f1)
+          COMPONENT_NAME=$(echo $TAG | cut -d/ -f2)
+          VERSION=$(echo $TAG | cut -d/ -f3)
+          
+          echo "component_type=$COMPONENT_TYPE" >> $GITHUB_OUTPUT
+          echo "component_name=$COMPONENT_NAME" >> $GITHUB_OUTPUT
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Extract SDK Version
+        id: sdk_version
+        run: |
+          SDK_VERSION=$(grep 'github.com/smithy-security/smithy/sdk' new-components/${{ steps.tag_info.outputs.component_type }}/${{ steps.tag_info.outputs.component_name }}/go.mod | awk '{print $2}')
```

Comment:
can we move this into a make target so that we can also run it locally and get the same result?

Reply:
Done